koku/masu/processor/tasks.py

```diff
@@ -315,6 +316,9 @@ def summarize_reports(reports_to_summarize, queue_name=None, manifest_list=None)
         ends = []
         if report and report.get("provider_type") in [Provider.PROVIDER_GCP, Provider.PROVIDER_GCP_LOCAL]:
             reports_deduplicated += deduplicate_reports_for_gcp(report_list)
+        elif report and report.get("provider_type") in [Provider.PROVIDER_OCI, Provider.PROVIDER_OCI_LOCAL]:
+            manifest_list = []
+            reports_deduplicated += deduplicate_reports_for_oci(report_list)
```

Comment:
This may not be true in newer Python versions, but using the `+=` operator uses more memory than the `.extend()` and `.append()` methods. This is because Python first has to build both lists is memory before combining them instead of modifying the existing list.
```suggestion
            reports_deduplicated.extend(deduplicate_reports_for_oci(report_list))
```

Reply:
Here is a rough example of how to keep the complexity of this method down by using a map to get the correct deduplicating function.
```python
        dedupe_func_map = {
            Provider.PROVIDER_GCP: deduplicate_reports_for_gcp,
            Provider.PROVIDER_GCP_LOCAL: deduplicate_reports_for_gcp,
            Provider.Provider_OCI: deduplicate_reports_for_oci,
            Provider.Provider_OCI_LOCAL: deduplicate_reports_for_oci,
        }
        provider_type = report.get("provider_type")
        if report and provider_type in dedupe_func_map:
            dedupe_func = dedupe_func_map.get(provider_type)
            reports_deduplicated.extend(dedupe_func())
```